src/nvim/api/private/helpers.c

```diff
@@ -615,15 +615,34 @@ bool object_to_vim(Object obj, typval_T *tv, Error *err)
     case kObjectTypeBuffer:
     case kObjectTypeWindow:
     case kObjectTypeTabpage:
-    case kObjectTypeInteger:
-      if (obj.data.integer > INT_MAX || obj.data.integer < INT_MIN) {
+    case kObjectTypeInteger: {
```

Comment:
Yes, I think that would be the best. (One could consider `int64_t buffer` but that would be confusing with the type difference.)